iron-list.html

```diff
@@ -1870,15 +1872,35 @@
         }
       }
     },
-
-    _didMoveUp: function() {
-      this._focusPhysicalItem(this._focusedIndex - 1);
-    },
-
-    _didMoveDown: function(e) {
+    // When `grid === true` the `downArrow` goes forward `this._itemsPerRow` items
+    _didMoveBackward: function(e, step) {
+      step = step || (!this.grid) ? 1 : this._itemsPerRow;
+      this._focusPhysicalItem(this._focusedIndex - step);
+    },
+    // When `grid === true` the `upArrow` goes backward `this._itemsPerRow` items
+    _didMoveForward: function(e, step) {
+      step = step || (!this.grid) ? 1 : this._itemsPerRow;
+      var destination = this._focusedIndex + step;
+	  // When `grid === true` move focus down one row, either
+	  // directly below or the last item of the list.
+	  if (this.grid && this._focusedIndex < (this.items.length - (this.items.length % this._itemsPerRow))) {
+		  destination = Math.min(destination, this.items.length - 1);
+	  }
       // disable scroll when pressing the down key
-      e.detail.keyboardEvent.preventDefault();
-      this._focusPhysicalItem(this._focusedIndex + 1);
+      if (e.detail.keyboardEvent.keyCode == 40) e.detail.keyboardEvent.preventDefault();
+      this._focusPhysicalItem(destination);
+    },
+    // When in grid mode, the use of `leftArrow` changes the focus based on the text direction
+    _didMoveLeft: function(e) {
+      if (!this.grid) return;
+      if (this._isRTL) this._didMoveForward(e, 1);
```

Comment:
Instead of calling another event handler (`_didMoveBackward`), just inline what it does here (and avoid overloading/remove the second param to `_didMoveBackward`):

```js
this._focusPhysicalItem(this._focusedIndex + (this._isRTL ? 1 : -1));
```